backend/fms_core/template_importer/importers/library_preparation.py

```diff
@@ -0,0 +1,103 @@
+import copy
+
+from fms_core.models import PropertyType, Protocol
+from ._generic import GenericImporter
+from fms_core.template_importer.row_handlers.library_preparation import LibraryPreparationRowHandler, LibraryBatchRowHandler
+from fms_core.templates import LIBRARY_PREPARATION_TEMPLATE
+from collections import defaultdict
+from .._utils import float_to_decimal_and_none, input_to_date_and_none
+
+PROPERTIES_STARTING_INDEX = 5
+
+class LibraryPreparationImporter(GenericImporter):
+    SHEETS_INFO = LIBRARY_PREPARATION_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.initialize_data_for_template()
+        self.properties_starting_index = PROPERTIES_STARTING_INDEX
+
+    def initialize_data_for_template(self):
+        # Get protocol for Library Preparation
+        protocol = Protocol.objects.get(name='Library Preparation')
+
+        self.preloaded_data = {'protocol': protocol, 'process_properties': {}}
+
+        # Preload PropertyType objects for this protocol in a dictionary for faster access
+        try:
+            self.preloaded_data['process_properties'] = \
+                {o.name: {'property_type_obj': o} for o in list(PropertyType.objects.filter(object_id=protocol.id))}
+        except Exception as e:
+            self.base_errors.append(f"Property Type could not be found. {e}")
+
+    def import_template_inner(self):
+        """
+            LIBRARY BATCH SHEET
+        """
+        library_batch_sheet = self.sheets['Library Batch']
+
+        # Iterate through libraries rows
+        library_batch_rows_data = defaultdict(list)
+        for row_id, row in enumerate(library_batch_sheet.rows):
+            library_batch_dict = {}
+            process_properties = self.preloaded_data['process_properties']
```

Comment:
This may be an error (i am aware that it is the same in the experiment_run ...). If we create more than one process the process_properties will be modified in between each iteration leading to false content for empty rows. This would need to be tested. I see you did a deepcopy lower, fixing this would require a deepcopy first to create a new memory reference then you could just pass off the reference at the end (if we test it and it does fail we'll need to check the experiment_run code too).